src/classes/BDI_DataImportFLSService.cls

```diff
@@ -0,0 +1,164 @@
+    * Neither the name of Salesforce.org nor the names of
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2019
+* @group Batch Data Import
+* @group-content ../../ApexDocContent/BatchDataImport.htm
+* @description Service class used to validate
+* Field Level Security of the target fields mapped in the data import process.
+*/
+public class BDI_DataImportFLSService {
+    Set<AccessLevel> accessLevels;
+    Map<SObjectField, BDI_TargetFields> targetFieldsBySObjectFields;
+    List<DataImport__c> dataImportsValidated;
+    List<DataImport__c> dataImportsInvalidated;
+    Map<Id, List<SObjectFieldWrapper>> invalidFieldsByDataImportId;
+
+    public enum AccessLevel {
+        VALIDATE_READ, VALIDATE_CREATE, VALIDATE_UPDATE
+    }
+
+    public BDI_DataImportFLSService(List<DataImport__c> dataImports,
+            BDI_FieldMapping fieldMapping,
+            Set<AccessLevel> accessLevels){
+        this.accessLevels = accessLevels;
+        this.targetFieldsBySObjectFields = fieldMapping.getTargetFieldsBySourceField();
+        this.dataImportsValidated = new List<DataImport__c>();
+        this.dataImportsInvalidated = new List<DataImport__c>();
+        this.invalidFieldsByDataImportId = new Map<Id, List<SObjectFieldWrapper>>();
+        validateTargetMappingFieldLevelSecurity(dataImports);
+    }
+
+    public List<DataImport__c> getValidRecords() {
+        return dataImportsValidated;
+    }
+
+    public List<DataImport__c> getInvalidRecords() {
+        return dataImportsInvalidated;
+    }
+
+    public List<String> getQualifiedInvalidFieldNamesById(Id dataImportId) {
+        List<String> qualifiedInvalidFieldNames = new List<String>();
+        for (SObjectFieldWrapper fieldWrapper :
+                invalidFieldsByDataImportId.get(dataImportId)) {
+            qualifiedInvalidFieldNames.add(fieldWrapper.getQualifiedName());
+        }
+        return qualifiedInvalidFieldNames;
+    }
+
+    void validateTargetMappingFieldLevelSecurity(List<DataImport__c> dataImportsInitial) {
```

Comment:
This method should specify `private` or `public`

Reply:
@kenlewis-sfdo ?

Reply:
My understanding is that method and member variable access is private by default.  @force2b Just to clarify, would this need to be changed for accessibility reasons, or just so that we are being explicit about the access level?

Reply:
Just the latter. I don't think we have any other methods or vars in the product that do specify the access level. Plus, I would expect this to be something that security review would call out.